catboost/python-package/catboost/core.py

```diff
@@ -1215,18 +1215,30 @@ def fit(self, X, y=None, cat_features=None, pairs=None, sample_weight=None, grou
                          save_snapshot, snapshot_file, snapshot_interval)
 
     def _predict(self, data, prediction_type, ntree_start, ntree_end, thread_count, verbose):
+        single_flag = False
         verbose = verbose or self.get_param('verbose')
         if verbose is None:
             verbose = False
         if not self.is_fitted():
             raise CatboostError("There is no trained model to use predict(). Use fit() to train model. Then use predict().")
+        if data is None:
+            raise CatboostError("Data to predict must be initialized")
         if not isinstance(data, Pool):
-            data = Pool(
-                data=data,
-                cat_features=self._get_cat_feature_indices() if not isinstance(data, FeaturesData) else None
-            )
-        elif not np.all(set(self._get_cat_feature_indices()).issubset(data.get_cat_feature_indices())):
-            raise CatboostError("Data cat_features in predict()={} are not equal data cat_features in fit()={}.".format(data.get_cat_feature_indices(), self._get_cat_feature_indices()))
+            if isinstance(data, ARRAY_TYPES):
+                if isinstance(data[0], ARRAY_TYPES):
+                    data = Pool(
+                        data=data,
+                        cat_features=self._get_cat_feature_indices() if not isinstance(data, FeaturesData) else None
+                    )
+                else:
+                    data = [data]
+                    single_flag = True
+                    data = Pool(
+                        data=data,
+                        cat_features=self._get_cat_feature_indices() if not isinstance(data, FeaturesData) else None
+                    )
+            elif not isinstance(data, FeaturesData):
```

Comment:
Please revert the change in the elif branch.
Cat features passed to predict() must be a subset of cat features passed to fit().